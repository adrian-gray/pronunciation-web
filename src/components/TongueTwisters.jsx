import React from "react";

import { capitalise } from "./../utils/utils";

import Paper from "@material-ui/core/Paper";
import Typography from "@material-ui/core/Typography";
import { makeStyles } from "@material-ui/core/styles";
import { withTheme } from "@material-ui/styles";

const useStyles = makeStyles(theme => ({
  headspace: theme.headspace,
  largeText: theme.largeText,
  spacing: {
    padding: "1rem",
    margin: "1rem"
  }
}));

function TongueTwisters(props) {
  const classes = useStyles(props);
  const { tag, tongueTwisters } = props;

  const twisters = tongueTwisters.map((line, index) => (
    <Paper className={classes.spacing} key={index}>
      <Typography>{line}</Typography>
    </Paper>
  ));

  return (
    <div className={classes.headspace}>
      <Typography variant="h5" gutterBottom>
        {`${capitalise(tag)} Tongue Twisters`}
      </Typography>
      <Typography>
        {`Enjoy the challenge of Tongue Twisters. There are three tongue twisters to perfect the ${tag} sound. Each tongue twister consists of commonly used English words and phrases for you to practice the ${tag} sound. How fast can you say each ${tag} tongue twister? Try to say each one as many times as you can, no mistakes. Record yourself and post your attempt to enter the challenge.`}
      </Typography>
      {twisters}
    </div>
  );
}

export default withTheme(TongueTwisters);
