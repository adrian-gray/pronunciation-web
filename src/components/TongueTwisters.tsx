import React from "react";

import Container from "react-bootstrap/Container";

import { capitalise } from "./../utils/utils";

export default props => {
  const { tag, tongueTwisters } = props;

  const twisters = tongueTwisters.map((line, index) => (
    <div className="personal-space sub-canvas" key={index}>
      <p>{line}</p>
    </div>
  ));

  return (
    <div className="head-space">
      <h3>{`${capitalise(tag)} Tongue Twisters`}</h3>
      <p>
        {`Enjoy the challenge of Tongue Twisters. There are three tongue twisters to perfect the ${tag} sound. Each tongue twister consists of commonly used English words and phrases for you to practice the ${tag} sound. How fast can you say each ${tag} tongue twister? Try to say each one as many times as you can, no mistakes. Record yourself and post your attempt to enter the challenge.`}
      </p>
      {twisters}
    </div>
  );
};
