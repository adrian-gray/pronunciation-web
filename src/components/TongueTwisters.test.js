import React from "react";
import { render, cleanup } from "@testing-library/react";

import TongueTwisters from "./TongueTwisters";

afterEach(cleanup);

test("TongueTwisters", async () => {
  const props = {
    tag: "B!",
    tongueTwisters: [
      "The cat can't cope crying",
      "My mum must make melon molasses"
    ]
  };

  const { container } = render(<TongueTwisters {...props} />);

  expect(container).toMatchSnapshot();
});
