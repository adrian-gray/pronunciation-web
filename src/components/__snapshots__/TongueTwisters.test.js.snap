// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TongueTwisters 1`] = `
<div>
  <div
    class="makeStyles-headspace-1"
  >
    <h5
      class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
    >
      B! Tongue Twisters
    </h5>
    <p
      class="MuiTypography-root MuiTypography-body1"
    >
      Enjoy the challenge of Tongue Twisters. There are three tongue twisters to perfect the B! sound. Each tongue twister consists of commonly used English words and phrases for you to practice the B! sound. How fast can you say each B! tongue twister? Try to say each one as many times as you can, no mistakes. Record yourself and post your attempt to enter the challenge.
    </p>
    <div
      class="MuiPaper-root MuiPaper-elevation1 makeStyles-spacing-3 MuiPaper-rounded"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        The cat can't cope crying
      </p>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation1 makeStyles-spacing-3 MuiPaper-rounded"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        My mum must make melon molasses
      </p>
    </div>
  </div>
</div>
`;
